### ISSUE TITLE
Xiphos crashes on launch

### ISSUE THREAD

**Issue by zrlaszlo** [crosswire/xiphos#1030]
Xiphos has been working on my system flawlessly for quite a while (Win10 Pro x64, Ryzen 7). After having installed yesterday Microsoft updates KB4556799 and KB4552931 Xiphos started crashing on launch: the splash screen appears, then the program window behind, but as soon as the launch screen is done, they both (launch screen and program window) disappear into thin air whether I run the program as and administrator or not. I upgraded to Xiphos version 4.1.2 x64, the problem remains. Xiphos is my favourite Bible software, what should I do?
Many thanks!

**Comment by alerque** [comment]
Have you tried upgrading to 4.2.1? Is was that what you meant to type instead of "4.1.2"?

**Comment by zrlaszlo** [comment]
Sorry for the typo, I meant 4.2.1, the latest build. Before I was running the build prior to 4.2.1 (can't remember the version number). My problems started with that one. I upgraded to the newest build, 4.2.1 and I the problem persists. Currently the program is unusable in my context.

**Comment by karlkleinpaste** [comment]
i don't know what to tell you about this. on the one win10 machine (real hardware) that i've got, it also shows having ...931 and ...799 installed, yet xiphos starts and runs immediately, no hesitation at all. are there any other recent differences to account for?

**Comment by zrlaszlo** [comment]
I updated some system drivers such as chipset and network adapter drivers. There were no other changes made to the system as far as I know. This afternoon I uninstalled Xiphos completely and reinstalled it. It does the same as described above. Still looking for a solution...

**Comment by ildar** [comment]
Running in console may show you something

**Comment by zrlaszlo** [comment]
> Running in console may show you something

Do you mean this? (see picture)
![Xiphos_console](https://user-images.githubusercontent.com/65437490/82143252-27726900-984b-11ea-9814-04c2e9858c56.jpg)

The outcome is the same as before.

**Comment by ildar** [comment]
Ain't no Linux 😞

**Comment by ildar** [comment]
Actually system journal Apps should have something

**Comment by karlkleinpaste** [comment]
no, "console mode" means a build with the console still enabled.  normally, windows graphical applications are built so as to silence the console entirely.
pick up this:
http://ftp.xiphos.org/sword/xiphos/xiphos-4.2.1.2-win64.exe
after install, start it from a cmd window. you may learn something useful.

**Comment by zrlaszlo** [comment]
I installed and ran the program above. The error mesage reads:

Error loading plugin: 'C:\Program Files\Crosswire\xiphos\lib\enchant\libenchant_myspell.dll': The specified module could not be found.

What is strange to me is that the file xiphos claims not to find is still there – see second attachment. There is a similar file in the \bin directory, should I replace the missing file from there?

**Comment by zrlaszlo** [comment]
Actually I did. Now the program does not return any kind of error message whatsoever, but sill crashes towards the end of the splash screen. Where to look for to find a clue?

**Comment by ildar** [comment]
the last thing comes to my mind is running with gdb or maybe with strace
analog: https://stackoverflow.com/questions/3847745/systrace-for-windows
Maybe Karl has better ideas.

**Comment by zrlaszlo** [comment]
I took a look today in the MS EventViewer and the following erorrs tied to every Xiphos startup (the part down to the dashed line is always the same, the rest of it may vary). Cen someone point out the problem? thank you!
Here is the error report
: 
Faulting application name: xiphos.exe, version: 0.0.0.0, time stamp: 0x00000000
Faulting module name: libwebkitgtk-1.0-0.dll, version: 0.0.0.0, time stamp: 0x00000000
Exception code: 0xc0000005
-----
Fault offset: 0x000000000085a38f
Faulting process ID: 0x190c
Faulting application start time: 0x01d62ce6fc0bbfa2
Faulting application path: C:\Program Files (x86)\CrossWire\Xiphos\bin\xiphos.exe
Faulting module path: C:\Program Files (x86)\CrossWire\Xiphos\bin\libwebkitgtk-1.0-0.dll
Report ID: 4a53ba18-b24b-46d8-9db7-a3bf9675fd7c
Faulting package full name: 
Faulting package-relative application ID:

**Comment by karlkleinpaste** [comment]
a bit of googling tells me that Exception code: 0xc0000005 is an "access violation," or what we in linux-land would call a segfault. probably some null pointer was accessed, and faceplant.

the idea that webkit is causing this is ... extraordinarily suspect. WK has been our display driver (it is responsible for all the HTML-driven formatting of all the textual display widgets) for some years now, and particularly in windows since v3.1.4 or so.

i cringe to suggest this, but...  there is a gdb (debugger) in xiphos' bin directory.  invoke xiphos indirectly using gdb.  that means, instead of "xiphos", run "gdb xiphos.exe" (you may have to give it a complete path to reach the xiphos.exe binary; i don't know), and after you get its self-announcement and a "(gdb)" prompt, say simply "r" for run.  it will do its thing for a while and (we presume) crash, at which point you will get an announcement by gdb of what went wrong, and a new prompt.  at that point, say "bt" (backtrace) and save what it tells you, but not more than the top 15 or so frames. (they're numbered.)  the process is still alive but suspended as it approaches death; just kill it with "k" and quit out of gdb.

gdb's backtrace is likely the end of the line for the help we can provide.

ultimately, this may be a problem of some corrupted module in your Sword area.  that could plausibly lead to such a drastic formatting failure that WK would choke.  but i doubt it.

**Comment by zrlaszlo** [comment]
Dear Karl, I followed your procedure, it was quite easy. Below you find the entire output, I hope it will shed some light on what is going on. Thank you for your concern!

(gdb) r
Starting program: c:\Program Files\Crosswire\xiphos\bin\xiphos.exe
[New Thread 4060.0x4640]
[New Thread 4060.0x2318]
[New Thread 4060.0x15d0]
[New Thread 4060.0x5208]
[New Thread 4060.0x4a3c]
[New Thread 4060.0x4118]
[New Thread 4060.0x532c]
[New Thread 4060.0x2324]
[New Thread 4060.0x3304]
[New Thread 4060.0x1b54]
[New Thread 4060.0x26d8]
[New Thread 4060.0x42e8]
[New Thread 4060.0x2010]
[New Thread 4060.0x3a90]
[New Thread 4060.0x4808]
[New Thread 4060.0x480c]
[New Thread 4060.0x4b00]

Thread 1 received signal SIGSEGV, Segmentation fault.
0x00000000017aa58f in WebCore::RenderStyle::computedLineHeight(WebCore::RenderView*) const ()
   from c:\Program Files\Crosswire\xiphos\bin\libwebkitgtk-1.0-0.dll
(gdb) bt
#0  0x00000000017aa58f in WebCore::RenderStyle::computedLineHeight(WebCore::RenderView*) const ()
   from c:\Program Files\Crosswire\xiphos\bin\libwebkitgtk-1.0-0.dll
#1  0x0000000001621cac in WebCore::RenderBlock::lineHeight(bool, WebCore::LineDirectionMode, WebCore::LinePositionMode) const () from c:\Program Files\Crosswire\xiphos\bin\libwebkitgtk-1.0-0.dll
#2  0x0000000001621d3e in WebCore::RenderBlock::minLineHeightForReplacedRenderer(bool, WebCore::LayoutUnit) const
    () from c:\Program Files\Crosswire\xiphos\bin\libwebkitgtk-1.0-0.dll
#3  0x000000000177dbe9 in WebCore::LineWidth::updateAvailableWidth(WebCore::LayoutUnit) ()
   from c:\Program Files\Crosswire\xiphos\bin\libwebkitgtk-1.0-0.dll
#4  0x000000000177b5f8 in WebCore::LineBreaker::nextSegmentBreak(WebCore::BidiResolver<WebCore::InlineIterator, WebCore::BidiRun>&, WebCore::LineInfo&, WebCore::RenderTextInfo&, WebCore::FloatingObject*, unsigned int, WTF::Vector<WebCore::WordMeasurement, 64ull, WTF::CrashOnOverflow>&) ()
   from c:\Program Files\Crosswire\xiphos\bin\libwebkitgtk-1.0-0.dll
#5  0x000000000177daf8 in WebCore::LineBreaker::nextLineBreak(WebCore::BidiResolver<WebCore::InlineIterator, WebCore::BidiRun>&, WebCore::LineInfo&, WebCore::RenderTextInfo&, WebCore::FloatingObject*, unsigned int, WTF::Vector<WebCore::WordMeasurement, 64ull, WTF::CrashOnOverflow>&) ()
   from c:\Program Files\Crosswire\xiphos\bin\libwebkitgtk-1.0-0.dll
#6  0x000000000163a688 in WebCore::RenderBlockFlow::layoutRunsAndFloatsInRange(WebCore::LineLayoutState&, WebCore::BidiResolver<WebCore::InlineIterator, WebCore::BidiRun>&, WebCore::InlineIterator const&, WebCore::BidiStatus const&, unsigned int) () from c:\Program Files\Crosswire\xiphos\bin\libwebkitgtk-1.0-0.dll
#7  0x000000000163d35a in WebCore::RenderBlockFlow::layoutRunsAndFloats(WebCore::LineLayoutState&, bool) ()
   from c:\Program Files\Crosswire\xiphos\bin\libwebkitgtk-1.0-0.dll
#8  0x000000000163e253 in WebCore::RenderBlockFlow::layoutLineBoxes(bool, WebCore::LayoutUnit&, WebCore::LayoutUnit&) () from c:\Program Files\Crosswire\xiphos\bin\libwebkitgtk-1.0-0.dll
#9  0x00000000016319f8 in WebCore::RenderBlockFlow::layoutBlock(bool, WebCore::LayoutUnit) ()
   from c:\Program Files\Crosswire\xiphos\bin\libwebkitgtk-1.0-0.dll
#10 0x000000000160d84c in WebCore::RenderBlock::layout() ()
   from c:\Program Files\Crosswire\xiphos\bin\libwebkitgtk-1.0-0.dll
#11 0x000000000162e42a in WebCore::RenderBlockFlow::layoutBlockChild(WebCore::RenderBox&, WebCore::RenderBlockFlow::MarginInfo&, WebCore::LayoutUnit&, WebCore::LayoutUnit&) ()
   from c:\Program Files\Crosswire\xiphos\bin\libwebkitgtk-1.0-0.dll
#12 0x000000000163054b in WebCore::RenderBlockFlow::layoutBlockChildren(bool, WebCore::LayoutUnit&) ()
   from c:\Program Files\Crosswire\xiphos\bin\libwebkitgtk-1.0-0.dll
#13 0x0000000001631f9c in WebCore::RenderBlockFlow::layoutBlock(bool, WebCore::LayoutUnit) ()
   from c:\Program Files\Crosswire\xiphos\bin\libwebkitgtk-1.0-0.dll
#14 0x000000000160d84c in WebCore::RenderBlock::layout() ()
   from c:\Program Files\Crosswire\xiphos\bin\libwebkitgtk-1.0-0.dll
#15 0x000000000162e42a in WebCore::RenderBlockFlow::layoutBlockChild(WebCore::RenderBox&, WebCore::RenderBlockFlow::MarginInfo&, WebCore::LayoutUnit&, WebCore::LayoutUnit&) ()
   from c:\Program Files\Crosswire\xiphos\bin\libwebkitgtk-1.0-0.dll
#16 0x000000000163054b in WebCore::RenderBlockFlow::layoutBlockChildren(bool, WebCore::LayoutUnit&) ()
   from c:\Program Files\Crosswire\xiphos\bin\libwebkitgtk-1.0-0.dll
#17 0x0000000001631f9c in WebCore::RenderBlockFlow::layoutBlock(bool, WebCore::LayoutUnit) ()
   from c:\Program Files\Crosswire\xiphos\bin\libwebkitgtk-1.0-0.dll
#18 0x000000000160d84c in WebCore::RenderBlock::layout() ()
   from c:\Program Files\Crosswire\xiphos\bin\libwebkitgtk-1.0-0.dll
#19 0x000000000162e42a in WebCore::RenderBlockFlow::layoutBlockChild(WebCore::RenderBox&, WebCore::RenderBlockFlow::MarginInfo&, WebCore::LayoutUnit&, WebCore::LayoutUnit&) ()
   from c:\Program Files\Crosswire\xiphos\bin\libwebkitgtk-1.0-0.dll
#20 0x000000000163054b in WebCore::RenderBlockFlow::layoutBlockChildren(bool, WebCore::LayoutUnit&) ()
   from c:\Program Files\Crosswire\xiphos\bin\libwebkitgtk-1.0-0.dll
#21 0x0000000001631f9c in WebCore::RenderBlockFlow::layoutBlock(bool, WebCore::LayoutUnit) ()
   from c:\Program Files\Crosswire\xiphos\bin\libwebkitgtk-1.0-0.dll
#22 0x000000000160d84c in WebCore::RenderBlock::layout() ()
   from c:\Program Files\Crosswire\xiphos\bin\libwebkitgtk-1.0-0.dll
#23 0x00000000017674f6 in WebCore::RenderView::layoutContent(WebCore::LayoutState const&) ()
   from c:\Program Files\Crosswire\xiphos\bin\libwebkitgtk-1.0-0.dll
#24 0x00000000017677d4 in WebCore::RenderView::layout() ()
   from c:\Program Files\Crosswire\xiphos\bin\libwebkitgtk-1.0-0.dll
#25 0x00000000015a825b in WebCore::FrameView::layout(bool) ()
   from c:\Program Files\Crosswire\xiphos\bin\libwebkitgtk-1.0-0.dll
#26 0x00000000011d3456 in WebCore::Document::implicitClose() ()
   from c:\Program Files\Crosswire\xiphos\bin\libwebkitgtk-1.0-0.dll
#27 0x00000000015058cc in WebCore::FrameLoader::checkCompleted() ()
   from c:\Program Files\Crosswire\xiphos\bin\libwebkitgtk-1.0-0.dll
#28 0x0000000001505975 in WebCore::FrameLoader::finishedParsing() ()
   from c:\Program Files\Crosswire\xiphos\bin\libwebkitgtk-1.0-0.dll
#29 0x00000000011d4f9c in WebCore::Document::finishedParsing() ()
   from c:\Program Files\Crosswire\xiphos\bin\libwebkitgtk-1.0-0.dll
#30 0x00000000013db3c2 in WebCore::HTMLDocumentParser::prepareToStopParsing() ()
---Type <return> to continue, or q <return> to quit---

**Comment by zrlaszlo** [comment]
This may also be helpful...

   from c:\Program Files\Crosswire\xiphos\bin\libwebkitgtk-1.0-0.dll
#31 0x00000000013d86b8 in WebCore::HTMLDocumentParser::finish() ()
   from c:\Program Files\Crosswire\xiphos\bin\libwebkitgtk-1.0-0.dll
#32 0x00000000014fd882 in WebCore::DocumentWriter::end() ()
   from c:\Program Files\Crosswire\xiphos\bin\libwebkitgtk-1.0-0.dll
#33 0x00000000014f52ef in WebCore::DocumentLoader::finishedLoading(double) ()
   from c:\Program Files\Crosswire\xiphos\bin\libwebkitgtk-1.0-0.dll
#34 0x00000000014f56a7 in WebCore::DocumentLoader::continueAfterContentPolicy(WebCore::PolicyAction) ()
   from c:\Program Files\Crosswire\xiphos\bin\libwebkitgtk-1.0-0.dll
#35 0x00000000014f9111 in WebCore::DocumentLoader::responseReceived(WebCore::CachedResource*, WebCore::ResourceResponse const&) () from c:\Program Files\Crosswire\xiphos\bin\libwebkitgtk-1.0-0.dll
#36 0x00000000014f36ab in WebCore::DocumentLoader::handleSubstituteDataLoadNow(WebCore::Timer<WebCore::DocumentLoader>*) () from c:\Program Files\Crosswire\xiphos\bin\libwebkitgtk-1.0-0.dll
#37 0x0000000001033cf8 in WebCore::ThreadTimers::sharedTimerFiredInternal() [clone .part.7] ()
   from c:\Program Files\Crosswire\xiphos\bin\libwebkitgtk-1.0-0.dll
#38 0x000000000104bf36 in WebCore::sharedTimerTimeoutCallback(void*) ()
   from c:\Program Files\Crosswire\xiphos\bin\libwebkitgtk-1.0-0.dll
#39 0x00000000685f5253 in g_timeout_dispatch () from c:\Program Files\Crosswire\xiphos\bin\libglib-2.0-0.dll
#40 0x00000000685f470e in g_main_context_dispatch () from c:\Program Files\Crosswire\xiphos\bin\libglib-2.0-0.dll
#41 0x00000000685f4a68 in g_main_context_iterate.isra ()
   from c:\Program Files\Crosswire\xiphos\bin\libglib-2.0-0.dll
#42 0x00000000685f4c7d in g_main_context_iteration () from c:\Program Files\Crosswire\xiphos\bin\libglib-2.0-0.dll
#43 0x0000000061848341 in gtk_main_iteration () from c:\Program Files\Crosswire\xiphos\bin\libgtk-win32-2.0-0.dll
#44 0x000000000043da60 in xiphos!on_edit_activate2 ()
#45 0x00000000004394d4 in xiphos!gui_menu_prayerlist_popup ()
#46 0x000000000045455e in xiphos!on_edit_activate2 ()
#47 0x00000000004013c5 in ?? ()
#48 0x00000000004014fb in ?? ()
#49 0x00007fff93d77bd4 in KERNEL32!BaseThreadInitThunk () from C:\Windows\System32\kernel32.dll
#50 0x00007fff959ece51 in ntdll!RtlUserThreadStart () from C:\Windows\SYSTEM32\ntdll.dll
#51 0x0000000000000000 in ?? ()
Backtrace stopped: previous frame inner to this frame (corrupt stack?)

**Comment by zrlaszlo** [comment]
Some more news: I removed all files from the folder C:\Users\zrlas\AppData\Roaming\Sword and launched Xiphos. It launched without error and started the Module Manager. I added a number of modules, restarted Xiphos but same problem: crashed during splash screen. I repeated the process you described, and here is the backtrace log:

(gdb) r
Starting program: c:\Program Files\Crosswire\xiphos\bin\xiphos.exe
[New Thread 2680.0x2cc4]
[New Thread 2680.0x2568]
[New Thread 2680.0x268c]
[New Thread 2680.0x2684]
[New Thread 2680.0x1c5c]
[New Thread 2680.0x26b8]
[New Thread 2680.0x26bc]
[New Thread 2680.0x504]
[New Thread 2680.0x580]
[New Thread 2680.0xbcc]
[New Thread 2680.0x318]
[New Thread 2680.0x314]
[New Thread 2680.0x480]
[New Thread 2680.0x610]
[New Thread 2680.0x364]
[New Thread 2680.0x57c]
[New Thread 2680.0x894]

Thread 1 received signal SIGSEGV, Segmentation fault.
0x000000000169a58f in WebCore::RenderStyle::computedLineHeight(WebCore::RenderView*) const ()
   from c:\Program Files\Crosswire\xiphos\bin\libwebkitgtk-1.0-0.dll
(gdb) bt
#0  0x000000000169a58f in WebCore::RenderStyle::computedLineHeight(WebCore::RenderView*) const ()
   from c:\Program Files\Crosswire\xiphos\bin\libwebkitgtk-1.0-0.dll
#1  0x0000000001511cac in WebCore::RenderBlock::lineHeight(bool, WebCore::LineDirectionMode, WebCore::LinePositionMode) const () from c:\Program Files\Crosswire\xiphos\bin\libwebkitgtk-1.0-0.dll
#2  0x0000000001511d3e in WebCore::RenderBlock::minLineHeightForReplacedRenderer(bool, WebCore::LayoutUnit) const ()
   from c:\Program Files\Crosswire\xiphos\bin\libwebkitgtk-1.0-0.dll
#3  0x000000000166dbe9 in WebCore::LineWidth::updateAvailableWidth(WebCore::LayoutUnit) ()
   from c:\Program Files\Crosswire\xiphos\bin\libwebkitgtk-1.0-0.dll
#4  0x000000000166b5f8 in WebCore::LineBreaker::nextSegmentBreak(WebCore::BidiResolver<WebCore::InlineIterator, WebCore::BidiRun>&, WebCore::LineInfo&, WebCore::RenderTextInfo&, WebCore::FloatingObject*, unsigned int, WTF::Vector<WebCore::WordMeasurement, 64ull, WTF::CrashOnOverflow>&) () from c:\Program Files\Crosswire\xiphos\bin\libwebkitgtk-1.0-0.dll
#5  0x000000000166daf8 in WebCore::LineBreaker::nextLineBreak(WebCore::BidiResolver<WebCore::InlineIterator, WebCore::BidiRun>&, WebCore::LineInfo&, WebCore::RenderTextInfo&, WebCore::FloatingObject*, unsigned int, WTF::Vector<WebCore::WordMeasurement, 64ull, WTF::CrashOnOverflow>&) () from c:\Program Files\Crosswire\xiphos\bin\libwebkitgtk-1.0-0.dll
#6  0x000000000152a688 in WebCore::RenderBlockFlow::layoutRunsAndFloatsInRange(WebCore::LineLayoutState&, WebCore::BidiResolver<WebCore::InlineIterator, WebCore::BidiRun>&, WebCore::InlineIterator const&, WebCore::BidiStatus const&, unsigned int) () from c:\Program Files\Crosswire\xiphos\bin\libwebkitgtk-1.0-0.dll
#7  0x000000000152d35a in WebCore::RenderBlockFlow::layoutRunsAndFloats(WebCore::LineLayoutState&, bool) ()
   from c:\Program Files\Crosswire\xiphos\bin\libwebkitgtk-1.0-0.dll
#8  0x000000000152e253 in WebCore::RenderBlockFlow::layoutLineBoxes(bool, WebCore::LayoutUnit&, WebCore::LayoutUnit&)
    () from c:\Program Files\Crosswire\xiphos\bin\libwebkitgtk-1.0-0.dll
#9  0x00000000015219f8 in WebCore::RenderBlockFlow::layoutBlock(bool, WebCore::LayoutUnit) ()
   from c:\Program Files\Crosswire\xiphos\bin\libwebkitgtk-1.0-0.dll
#10 0x00000000014fd84c in WebCore::RenderBlock::layout() ()
   from c:\Program Files\Crosswire\xiphos\bin\libwebkitgtk-1.0-0.dll
#11 0x000000000151e42a in WebCore::RenderBlockFlow::layoutBlockChild(WebCore::RenderBox&, WebCore::RenderBlockFlow::MarginInfo&, WebCore::LayoutUnit&, WebCore::LayoutUnit&) ()
   from c:\Program Files\Crosswire\xiphos\bin\libwebkitgtk-1.0-0.dll
#12 0x000000000152054b in WebCore::RenderBlockFlow::layoutBlockChildren(bool, WebCore::LayoutUnit&) ()
---Type <return> to continue, or q <return> to quit---
   from c:\Program Files\Crosswire\xiphos\bin\libwebkitgtk-1.0-0.dll
#13 0x0000000001521f9c in WebCore::RenderBlockFlow::layoutBlock(bool, WebCore::LayoutUnit) ()
   from c:\Program Files\Crosswire\xiphos\bin\libwebkitgtk-1.0-0.dll
#14 0x00000000014fd84c in WebCore::RenderBlock::layout() ()
   from c:\Program Files\Crosswire\xiphos\bin\libwebkitgtk-1.0-0.dll
#15 0x000000000151e42a in WebCore::RenderBlockFlow::layoutBlockChild(WebCore::RenderBox&, WebCore::RenderBlockFlow::MarginInfo&, WebCore::LayoutUnit&, WebCore::LayoutUnit&) ()
   from c:\Program Files\Crosswire\xiphos\bin\libwebkitgtk-1.0-0.dll
#16 0x000000000152054b in WebCore::RenderBlockFlow::layoutBlockChildren(bool, WebCore::LayoutUnit&) ()
   from c:\Program Files\Crosswire\xiphos\bin\libwebkitgtk-1.0-0.dll
#17 0x0000000001521f9c in WebCore::RenderBlockFlow::layoutBlock(bool, WebCore::LayoutUnit) ()
   from c:\Program Files\Crosswire\xiphos\bin\libwebkitgtk-1.0-0.dll
#18 0x00000000014fd84c in WebCore::RenderBlock::layout() ()
   from c:\Program Files\Crosswire\xiphos\bin\libwebkitgtk-1.0-0.dll
#19 0x000000000151e42a in WebCore::RenderBlockFlow::layoutBlockChild(WebCore::RenderBox&, WebCore::RenderBlockFlow::MarginInfo&, WebCore::LayoutUnit&, WebCore::LayoutUnit&) ()
   from c:\Program Files\Crosswire\xiphos\bin\libwebkitgtk-1.0-0.dll
#20 0x000000000152054b in WebCore::RenderBlockFlow::layoutBlockChildren(bool, WebCore::LayoutUnit&) ()
   from c:\Program Files\Crosswire\xiphos\bin\libwebkitgtk-1.0-0.dll
#21 0x0000000001521f9c in WebCore::RenderBlockFlow::layoutBlock(bool, WebCore::LayoutUnit) ()
   from c:\Program Files\Crosswire\xiphos\bin\libwebkitgtk-1.0-0.dll
#22 0x00000000014fd84c in WebCore::RenderBlock::layout() ()
   from c:\Program Files\Crosswire\xiphos\bin\libwebkitgtk-1.0-0.dll
#23 0x00000000016574f6 in WebCore::RenderView::layoutContent(WebCore::LayoutState const&) ()
   from c:\Program Files\Crosswire\xiphos\bin\libwebkitgtk-1.0-0.dll
#24 0x00000000016577d4 in WebCore::RenderView::layout() ()
   from c:\Program Files\Crosswire\xiphos\bin\libwebkitgtk-1.0-0.dll
#25 0x000000000149825b in WebCore::FrameView::layout(bool) ()
   from c:\Program Files\Crosswire\xiphos\bin\libwebkitgtk-1.0-0.dll
---Type <return> to continue, or q <return> to quit---

**Comment by karlkleinpaste** [comment]
this sequence is insane.

44 0x000000000043da60 in xiphos!on_edit_activate2 ()
45 0x00000000004394d4 in xiphos!gui_menu_prayerlist_popup ()
46 0x000000000045455e in xiphos!on_edit_activate2 ()
47 0x00000000004013c5 in ?? ()

on_edit_activate2 is part of the process of beginning to edit an existing prayerlist/journal.
gui_menu_prayerlist_popup is the beginning of offering a new prayerlist/journal.
neither of these events is plausibly happening at start, nor do they mutually reference each other, that one could call the other which would call back.

i'm afraid that what this means is that at some point the backtrace is invalid. that can be caused by someone using a pointer so badly that it trashes the stack, for example. other possibilities exist.

bottom line is that i know no more now than i did 20 minutes ago.  this is a fundamental "cannot reproduce" problem.  you're the only person showing this problem, and you had it under both 4.1.0 and 4.2.1.  this can tell me only that it is your system, not xiphos, that is causing a problem. but on the win10 boxes i have, fully updated, xiphos starts just fine.

i don't know what else to offer.  i don't have any other diagnostic suggestions.

**Comment by ildar** [comment]
Joining this guessing game: do other applications crash at times? Maybe
Windows BSOD? if so, this is hardware problem.

**Comment by zrlaszlo** [comment]
Windows is running just fine without any issues. What I have noticed is that between the two backtraces there is a difference: in the second one there is no reference to the prayerlist Karl mentioned. Ï will try to roll back the display driver, other option does not come to my mind concerning my system.

**Comment by loveofthetruth** [comment]
I'm having the same problem with Xiphos 4.2.1 on windows 10 64 bit on a Lenovo ThinkPad e580 laptop, it crashes on startup after the splash screen closes and the program window comes up, it then disappears, I also used gdp to debug and this is what I got: 
C:\Program Files\CrossWire\Xiphos\bin>gdb xiphos.exe
GNU gdb (GDB) 8.0
Copyright (C) 2017 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.  Type "show copying"
and "show warranty" for details.
This GDB was configured as "x86_64-w64-mingw32".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<http://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
<http://www.gnu.org/software/gdb/documentation/>.
For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from xiphos.exe...(no debugging symbols found)...done.
(gdb) r
Starting program: C:\Program Files\CrossWire\Xiphos\bin\xiphos.exe 
[New Thread 3896.0xf28]
[New Thread 3896.0x353c]
[New Thread 3896.0x1b88]
[New Thread 3896.0x4788]
[New Thread 3896.0x18b8]
[New Thread 3896.0x38a4]
[New Thread 3896.0x24ac]
[New Thread 3896.0x1a74]
[New Thread 3896.0x2654]
[New Thread 3896.0x419c]
[New Thread 3896.0x3290]
[New Thread 3896.0x19d8]
[New Thread 3896.0x3288]
[New Thread 3896.0x24a8]
[New Thread 3896.0x3ce4]
[New Thread 3896.0x1210]
[New Thread 3896.0x3c2c]
[New Thread 3896.0x191c]

Thread 1 received signal SIGSEGV, Segmentation fault.
0x00000000017da58f in WebCore::RenderStyle::computedLineHeight(WebCore::RenderView*) const ()
   from C:\Program Files\CrossWire\Xiphos\bin\libwebkitgtk-1.0-0.dll

I have had the same problem on windows 7 32 on an older laptop with an older version of Xiphos but I can't remember the version, according to this debug output it looks like a webkit issue, how is it possible that I have had the issue on two different machines with two different versions of Xiphos, I have not had much look with Xiphos on Windows 7 or 10

**Comment by LAfricain** [comment]
@zrlaszlo Can you try the new release?

**Comment by karlkleinpaste** [comment]
closing for lack of input. countless changes in windows (incl introduction of win11) makes this seem not relevant.
